plugin/externalPluginEntry.go

```diff
@@ -532,44 +532,8 @@ func newStdoutDecodeErr(ctx context.Context, decodedThing string, reason error,
 	return newInvokeError(fmt.Sprintf("could not decode %v from stdout: %v", decodedThing, reason), inv)
 }
 
-func (e *externalPluginEntry) pluginName() string {
-	pluginName, _ := e.splitTypeID()
-	return pluginName
-}
-
-func (e *externalPluginEntry) rawTypeID() string {
-	_, rawTypeID := e.splitTypeID()
-	return rawTypeID
-}
-
-// splits the type ID into a tuple (<plugin_name>, <raw_type_id>)
-func (e *externalPluginEntry) splitTypeID() (string, string) {
-	substrings := strings.SplitN(e.TypeID(), "::", 2)
-	if len(substrings) < 2 {
-		// We can hit this case if only some of the entries specify type IDs.
-		// This is impossible in the (common) entry schema scenario because
-		// entry schema support always requires type IDs. So, it is not an
-		// error.
-		//
-		// If we hit this case, then the concept of a type ID doesn't make
-		// sense so we return empty here. Note that returning substrings[0]
-		// is not a good idea b/c the plugin author could namespace their
-		// type IDs with "::", so substrings[0] would contain that initial
-		// chunk.
-		return "__unknown__", ""
-	}
-	return substrings[0], substrings[1]
-}
-
-func namespace(pluginName string, typeID string) string {
-	if len(typeID) == 0 {
-		typeID = "unknown"
-	}
-	return pluginName + "::" + typeID
-}
-
-func (e *externalPluginEntry) unmarshalSchemaGraph(stdout []byte) (*linkedhashmap.Map, error) {
-	pluginName, rawTypeID := e.splitTypeID()
+func unmarshalSchemaGraph(e externalPlugin, stdout []byte) (*linkedhashmap.Map, error) {
```

Comment:
Refactored this so that the casing in pluginName would work properly if e was an externalPluginRoot.

Reply:
Specific reason is because when I create the schema graph in `Init`, the root's ID is empty so we'll hit the cases in https://github.com/ekinanp/wash/blob/namespace-coreplugin-typeids/plugin/entrySchema.go#L272-L286. If this is on externalPluginEntry, then it will hit the default branch (externalPluginEntry is an embedded type in externalPluginRoot). Making it take the entry as a param. fixes that.